crypto/s2n_libcrypto.c

```diff
@@ -141,18 +141,34 @@ bool s2n_libcrypto_is_boringssl()
 S2N_RESULT s2n_libcrypto_init(void)
 {
 #if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
-    RESULT_ENSURE(OSSL_PROVIDER_load(NULL, "default") != NULL, S2N_ERR_OSSL_LOAD_PROVIDER);
+    RESULT_ENSURE(OSSL_PROVIDER_load(NULL, "default") != NULL, S2N_ERR_OSSL_PROVIDER);
     #ifdef S2N_LIBCRYPTO_SUPPORTS_EVP_RC4
     /* needed to support RC4 algorithm
      * https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html
      */
-    RESULT_ENSURE(OSSL_PROVIDER_load(NULL, "legacy") != NULL, S2N_ERR_OSSL_LOAD_PROVIDER);
+    RESULT_ENSURE(OSSL_PROVIDER_load(NULL, "legacy") != NULL, S2N_ERR_OSSL_PROVIDER);
     #endif
 #endif
 
     return S2N_RESULT_OK;
 }
 
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+int s2n_libcrypto_cleanup_cb(OSSL_PROVIDER *provider, void *cbdata) {
+    return OSSL_PROVIDER_unload(provider);
+}
+
+S2N_RESULT s2n_libcrypto_cleanup(void) {
+    RESULT_GUARD_OSSL(OSSL_PROVIDER_do_all(NULL, *s2n_libcrypto_cleanup_cb, NULL), S2N_ERR_ATEXIT);
+
+    return S2N_RESULT_OK;
```

Comment:
Nit, kind of silly spacing
```suggestion
    return S2N_RESULT_OK;
```

Reply:
I think its fine. It separates the important code from the `return` bit. We do this quite often actualy:

https://github.com/aws/s2n-tls/blob/bc51f394b57c1b63bbd38d3a3d90f177e37cb561/crypto/s2n_libcrypto.c#L71-L73

Reply:
We usually don't do it for single line functions :)